sequence_python/models/ir_sequence.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2020 Sunflower IT (<https://sunflowerweb.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+import random
+import string
+import uuid
+
+from odoo import fields, models
+from odoo.tools import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+DEFAULT_PYTHON_CODE = "number_padded"
+
+
+class IrSequence(models.Model):
+    """
+    Inherit standard ir.sequence to let the possibility of using a Python formula
+    to calculate the sequence from input variables such as the sequence number.
+    This allows obfuscation of the order in which sequences are given out,
+    but can also be used for any other purpose.
+    """
+
+    _inherit = "ir.sequence"
+
+    # Python code
+    use_python_code = fields.Boolean(string="Use Python", default=False)
+    python_code = fields.Text(
+        string="Python expression",
+        groups="base.group_system",
+        default=DEFAULT_PYTHON_CODE,
+        help="Write Python code that generates the sequence body.",
+    )
+    python_code_preview = fields.Char("Preview", compute="_compute_python_code_preview")
+
+    def _get_python_eval_context(self, number_next):
+        """
+        Get the eval context to evaluate the Python code with.
+        The format is (variable name, description, value)
+        You can inherit this in your custom module.
+        :return: tuple
+        """
+        return {
+            "number": number_next[0],
```

Comment:
Getting `'int' object is not subscriptable` here coming out of a no_gap sequence with python_code `uuid.uuid4().hex`